### PR TITLE
IO misc refactoring & upgrades

### PR DESCRIPTION
**Pull request by joeflack4**
## Changes
IO misc refactoring & upgrades
- Update: get_mim_file(): Now adding .txt at the end of file_name param is optional. Usages updated to leave it off.
- Update: get_mim_file(): Now filters comments out before returning lines. For now, left code appearing later on which also filters out any lines starting with #.
- Update: get_mim_file(): Now correctly converts all .txt to .tsv in a more stable way, using new function convert_txt_to_tsv().

## Additional info
@twhetzel I want to do some analysis of `mimTitles` for #119. But I did not have a TSV to work with. This PR primarily fixes that for `mimTitles`, and also now (i) ensures every `.txt` has a readable `.tsv`, and (ii) future proofs code by replacing some hard-coded headers.
I may also need this as a prerequisite for #119 because it'll be easier to work with a dataframe/TSV for this.

### CODE REVIEW

**Comment by twhetzel** [comment]
@joeflack4 I don't see why these changes are needed. What was not working? I would not remove file extensions from filenames.

File: omim2obo/parsers/omim_txt_parser.py
Comment (joeflack4) [inline omim2obo/parsers/omim_txt_parser.py:1]: ## General overview
[@twhetzel:](https://github.com/monarch-initiative/omim/pull/123#issuecomment-2329675400):
> What was not working? 

I haven't been adding comment threads to my PRs, but I figured you'd probably want a brief summary. I wrote in [the OP](https://github.com/monarch-initiative/omim/pull/123#issue-2495648928):
> I want to do some analysis of `mimTitles` for #119. But I did not have a TSV to work with. This PR primarily fixes that for `mimTitles`, and also now (i) ensures every `.txt` has a readable `.tsv`, and (ii) future proofs code by replacing some hard-coded headers.

W/ these changes, I was able to do the analysis needed.

This has been a big pain point for a while. TSVs can be analyzed in Excel or `pandas`, but the native `.txt` files are hard to analyze.

[@twhetzel:](https://github.com/monarch-initiative/omim/pull/123#issuecomment-2329675400):
> I would not remove file extensions from filenames.

No worries; it doesn't. It creates a second set of TSV files which are strictly for analysis. It's been doing this for a while, but was not doing it correctly for many of the files.

Comment (joeflack4) [inline omim2obo/parsers/omim_txt_parser.py:1]: [@twhetzel:](https://github.com/monarch-initiative/omim/pull/123#issuecomment-2329675400):
> I would not remove file extensions from filenames.

No worries; it doesn't. It creates a second set of TSV files which are strictly for analysis. It's been doing this for a while, but was not doing it correctly for many of the files.

Comment (joeflack4) [reply omim2obo/parsers/omim_txt_parser.py:1]: Hmm, I think the only issue there is munging the top of the file's comments, and getting just the header to work. When you want to see the header and also do operations like check unique values, you need to delete the comments at the top and the bottom, and uncomment the header. And by default it re-downloads these files every time I run the program / restart the debugger.

Comment (twhetzel) [reply omim2obo/parsers/omim_txt_parser.py:1]: You can register as an individual for an API Key and get the mimTitles.txt file for analysis. The ".txt" file can be read in pandas.

Comment (twhetzel) [reply omim2obo/parsers/omim_txt_parser.py:1]: There are multiple lines above in omim2obo/main.py where you have dropped the ".txt" extension.

Comment (joeflack4) [reply omim2obo/parsers/omim_txt_parser.py:1]: That's just for the second set of files. These are the files that it creates in `data/`:
```
genemap2.tsv
genemap2.txt
mim2gene.tsv
mim2gene.txt
mimTitles.tsv
mimTitles.txt
morbidmap.tsv
morbidmap.txt
phenotypicSeries.tsv
phenotypicSeries.txt
```

The `.txt` ones are left unchanged and are what the codebase is currently processing. The `.tsv` ones have been there for a while too, but the header/comment fix was only working for 2 of them before this PR. They aren't used in the codebase; just there for analysis.

**Review by twhetzel — APPROVED** [review]
seems fine, possibly some other options, but it's ok

**Pull request merged**